crowbar_framework/lib/crowbar/backup/restore.rb

```diff
@@ -67,19 +152,33 @@ def restore_chef
                 )
                 proposal.properties = record.raw_data
                 proposal.save
-                SchemaMigration.run_for_bc(bc_name)
+                begin
+                  Rails.logger.debug "Migrating #{bc_name} schema"
+                  SchemaMigration.run_for_bc(bc_name)
+                rescue StandardError => e
+                  set_failed
+                  msg = I18n.t(
+                    ".installer.upgrades.restore.schema_migration_failed",
+                    bc_name: bc_name
+                  )
+                  Rails.logger.error("#{msg} -- #{e.message}")
+                  @status[:restore_chef] = {
+                    status: :conflict,
+                    msg: msg
+                  }
+                end
               else
                 record.save
               end
             end
           end
+
+          @status[:restore_chef] = { status: :ok, msg: "" }
```

Comment:
Doesn't this overwrite the status that is set when an error happens during Schema migration?